plugin/advancedsubscription/ajax/advsub.ajax.php

```diff
@@ -0,0 +1,248 @@
+                            'advancedsubscription/src/admin_view.php?s=' . $data['s'];
+                        // Send mails
+                        $result['mailIds'] = $plugin->sendMail($data, ADV_SUB_ACTION_STUDENT_REQUEST_NO_BOSS);
+                        // Check if mails were sent
+                        if (!empty($result['mailIds'])) {
+                            $result['error'] = false;
+                            $result['errorMessage'] = 'No error';
+                            $result['pass'] = true;
+                            // Check if exist an email to render
+                            if (isset($result['mailIds']['render'])) {
+                                // Render mail
+                                $message = MessageManager::get_message_by_id($result['mailIds']['render']);
+                                $message = str_replace(array('<br /><hr>', '<br />', '<br/>'), '', $message['content']);
+                                echo $message;
+                                exit;
+                            }
+                        }
+                    }
+                } else {
+                    // Student boss does exist
+                    // Get url to be accepted by boss
+                    $data['e'] = ADV_SUB_QUEUE_STATUS_BOSS_APPROVED;
+                    $data['student']['acceptUrl'] = $plugin->getQueueUrl($data);
+                    // Get url to be rejected by boss
+                    $data['e'] = ADV_SUB_QUEUE_STATUS_BOSS_DISAPPROVED;
+                    $data['student']['rejectUrl'] = $plugin->getQueueUrl($data);
+                    // Send mails
+                    $result['mailIds'] = $plugin->sendMail($data, ADV_SUB_ACTION_STUDENT_REQUEST);
+                    // Check if mails were sent
+                    if (!empty($result['mailIds'])) {
+                        $result['error'] = false;
+                        $result['errorMessage'] = 'No error';
+                        $result['pass'] = true;
+                        // Check if exist an email to render
+                        if (isset($result['mailIds']['render'])) {
+                            // Render mail
+                            $message = MessageManager::get_message_by_id($result['mailIds']['render']);
+                            $message = str_replace(array('<br /><hr>', '<br />', '<br/>'), '', $message['content']);
+                            echo $message;
+                            exit;
+                        }
+                    }
+                }
+            } else {
+                if (is_string($res)) {
+                    $result['errorMessage'] = $res;
+                } else {
+                    $result['errorMessage'] = 'User can not be subscribed';
+                }
+                $result['pass'] = false;
+            }
+
+            break;
+        case 'confirm':
+            // Check if is set new status
+            if (isset($data['e'])) {
```

Comment:
Avoid using of shortcuts like $data['e']
Use complete words: $data['error'],  $data['session'] etc